src/invoice/signature.rs

```diff
@@ -131,6 +132,18 @@ impl KeyEntry {
         let sig = key.sign(self.label.as_bytes());
         self.label_signature = Some(base64::encode(sig.to_bytes()));
     }
+    pub fn verify_label(self, key: PublicKey) -> anyhow::Result<()> {
+        match self.label_signature {
+            // If there is no label signature, then I guess we say everything's good.
+            None => Ok(()),
+            Some(txt) => {
+                let decoded_txt = base64::decode(txt.as_bytes())?;
```

Comment:
Nit: I don't think you need `as_bytes` here? `decode` takes `AsRef<[u8]>` which `str` implements